moshi/src/main/java/com/squareup/moshi/JsonUtf8Writer.java

```diff
@@ -91,7 +91,7 @@
   }
 
   @Override public JsonWriter endObject() throws IOException {
-    promoteValueToName = false;
+    promoteValueToNameFailureToken = null;
```

Comment:
that's what i'm unsure about.

the options i see:
1. Leave the code unchanged (only use the boolean field, revert my last commit on this PR). The error message will be generic and won't have the JSON token.
2. Change the field to a nullable JsonReader.Token, where null is the existing `false`, and the non-null value can be used in the error message.

I'm leaning toward 1 because of the diminished complexity.